x-pack/plugins/observability/public/pages/cases/helpers.ts

```diff
@@ -4,16 +4,91 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { useEffect, useState } from 'react';
+import { useEffect, useState, useCallback, useMemo } from 'react';
 import { isEmpty } from 'lodash';
+
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
 import { usePluginContext } from '../../hooks/use_plugin_context';
 import { TopAlert, parseAlert } from '../../pages/alerts/';
 import { useKibana } from '../../utils/kibana_react';
-import { Ecs } from '../../../../cases/common';
+import { BASE_RAC_ALERTS_API_PATH } from '../../../../rule_registry/common/constants';
+
+export const useFetchAlertData = (alertIds: string[]): [boolean, Record<string, unknown>] => {
+  const { http } = useKibana().services;
+  const [loading, setLoading] = useState(false);
+  const [alerts, setAlerts] = useState<Record<string, unknown> | undefined>(undefined);
+
+  const fetch = useCallback(
```

Comment:
alternatively you can write this fetch function out of the component, that accept parameters and returns a promise, then where you use it you await the promise and call `setAlerts` there.